src/Core/Resolvers/SqlQueryEngine.cs

```diff
@@ -205,15 +213,29 @@ public async Task<IActionResult> ExecuteAsync(StoredProcedureRequestContext cont
 
             // Open connection and execute query using _queryExecutor
             string queryString = queryBuilder.Build(structure);
-            JsonDocument? jsonDocument =
-                await queryExecutor.ExecuteQueryAsync(
-                    sqltext: queryString,
-                    parameters: structure.Parameters,
-                    dataReaderHandler: queryExecutor.GetJsonResultAsync<JsonDocument>,
-                    httpContext: _httpContextAccessor.HttpContext!,
-                    args: null,
-                    dataSourceName: dataSourceName);
+
+            if (await _featureManager.IsEnabledAsync("CachingPreview"))
+            {
```

Comment:
Could the SQL connection string specify the application intent as Read only or read write?
Should we base the decision of whether to cache or not on this insight? Or perhaps use it in addition to the feature flag in config

Reply:
Can you elaborate a bit more here, Ani? Are you suggesting we don't need a feature flag to turn on/off caching?

The application intent in the connection string is related to the level of permission which the user has, right? Even if this user has read-only connection, there might be some other user who is performing mutation on the data the first user is fetching. So if we based our decision on the application intent for the first user, he would get back stale data right?

Reply:
I may be failing to see the implications of the connection string declaring read vs read/write. As long as data can be read from the db, the data can be cached based on
- the runtime configuration settings defined
- other checks performed like ensuring MsSql SetSessionContext feature is not enabled.